### PR TITLE
fix: Basic auth plugin should run in rewrite phase.

### PR DESCRIPTION
**Pull request by jenskeiner**
### What this PR does / why we need it:
The `basic-auth` plugin, as an authentication plugin, should run in the rewrite phase, not the access phase.

### Pre-submission checklist:

* [x] Did you explain what problem does this PR solve? Or what new features have been added?
* [x] Have you added corresponding test cases?
* [ ] Have you modified the corresponding document?
* [x] Is this PR backward compatible? **If it is not backward compatible, please discuss on the [mailing list](https://github.com/apache/apisix/tree/master#community) first**

### CODE REVIEW

**Comment by jenskeiner** [comment]
As per comment from @spacewander, added a check for the consumer handling; see https://github.com/apache/apisix/pull/2859#discussion_r531816904.

File: apisix/init.lua [hunk 1]
```diff
@@ -513,6 +513,8 @@ function _M.http_access_phase()
                 api_ctx.consumer,
                 api_ctx
             )
+            core.log.debug("find consumer ", api_ctx.consumer.username,
```
Comment (spacewander) [inline apisix/init.lua:516]: IMHO, we can use `info` level error log, therefore we can check it without `log_level('debug')`. `debug` is too low to check it in the production if needed.

File: t/plugin/basic-auth.t [hunk 2]
```diff
@@ -189,6 +190,7 @@ Authorization: Basic Zm9vOmZvbwo=
 
 
 === TEST 8: verify
+--- yaml_config eval: $::yaml_config
```
Comment (spacewander) [inline t/plugin/basic-auth.t:193]: We don't need this line?

File: apisix/init.lua (hunk 1 above)
Comment (jenskeiner) [reply apisix/init.lua:516]: Sure, just thought we don't want to clutter the log too much in a prod setting where you might be running with `info` level.

File: t/plugin/basic-auth.t (hunk 2 above)
Comment (jenskeiner) [reply t/plugin/basic-auth.t:193]: Yep, will remove it. Was left over from my first attempt at changing the log level.

**Review by spacewander — APPROVED** [review]

**Review by moonming — APPROVED** [review]

**Comment by membphis** [comment]
@jenskeiner many thx, merged right now

**Pull request merged**